src/components/store/store.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useState,
+} from 'react';
+
+import { Todo } from '../../types/Todo';
+import { ProviderProps } from '../../types/ProviderProps';
+import { FilterOption } from '../../enum/FilterOption';
+
+type TodosContextType = {
+  todos: Todo[],
+  reciveTodos: (newTodos: Todo[]) => void,
+  setFilterSelected: (filter: FilterOption) => void,
+  filterSelected: FilterOption,
+  filteredTodos: Todo[],
+  toggleTodoCondition: (todoId: number) => void,
+};
+
+const TodosContext = createContext<TodosContextType>({
+  todos: [],
+  reciveTodos: () => {},
+  setFilterSelected: () => {},
+  filterSelected: FilterOption.All,
+  filteredTodos: [],
+  toggleTodoCondition: () => {},
+});
+
+export const TodosContextProvider: React.FC<ProviderProps> = ({
+  children,
+}) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [
+    filterSelected,
+    setFilterSelected,
+  ] = useState<FilterOption>(FilterOption.All);
+
+  const reciveTodos = useCallback(
+    (newTodos: Todo[]) => {
+      setTodos(newTodos);
+    },
+    [],
+  );
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterSelected) {
+      case FilterOption.Active:
+        return !todo.completed;
+
+      case FilterOption.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
```

Comment:
Filtering logic should be brought out as a separate helper outside of components to improve code readability